scripts/scil_compute_tracking_dipy.py

```diff
@@ -0,0 +1,265 @@
+from dipy.tracking import utils as track_utils
+
+import nibabel as nib
+from nibabel.streamlines import Field
+from nibabel.orientations import aff2axcodes
+from nibabel.streamlines.tractogram import LazyTractogram
+import numpy as np
+
+from scilpy.reconst.utils import (find_order_from_nb_coeff,
+                                  get_b_matrix, get_maximas)
+from scilpy.tracking.tools import get_theta
+from scilpy.io.utils import (create_header_from_anat,
+                             add_overwrite_arg, add_sh_basis_args, add_verbose,
+                             assert_inputs_exist, assert_outputs_exists)
+
+DETERMINISTIC = 'det'
+PROBABILISTIC = 'prob'
+EUDX = 'eudx'
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    p._optionals.title = 'Generic options'
+    p.add_argument('sh_file',
+                   help='Spherical harmonic file. \n'
+                   '(isotropic resolution, nifti, see --basis).')
+    p.add_argument('seed_file',
+                   help='Seeding mask (isotropic resolution, nifti).')
+    p.add_argument('mask_file',
+                   help='Seeding mask(isotropic resolution, nifti).\n' +
+                   'Tracking will stop outside this mask')
+    p.add_argument('output_file',
+                   help='Streamline output file (must be trk or tck).')
+
+    track_g = p.add_argument_group('Tracking options')
+    track_g.add_argument(
+        '--algo', default='prob', choices=['det', 'prob'],
+        help='Algorithm to use (must be "det" or "prob"). [%(default)s]')
+    track_g.add_argument(
+        '--step', dest='step_size', type=float, default=0.5,
+        help='Step size in mm. [%(default)s]')
+    track_g.add_argument(
+        '--min_length', type=float, default=10.,
+        help='Minimum length of a streamline in mm. [%(default)s]')
+    track_g.add_argument(
+        '--max_length', type=float, default=300.,
+        help='Maximum length of a streamline in mm. [%(default)s]')
+    track_g.add_argument(
+        '--theta', type=float,
+        help='Maximum angle between 2 steps. ["eudx"=60, det"=45, "prob"=20]')
+    track_g.add_argument(
+        '--sfthres', dest='sf_threshold', type=float, default=0.1,
+        help='Spherical function relative threshold. [%(default)s]')
+    add_sh_basis_args(track_g)
+
+    seed_group = p.add_argument_group(
+        'Seeding options',
+        'When no option is provided, uses --npv 1.')
+    seed_sub_exclusive = seed_group.add_mutually_exclusive_group()
+    seed_sub_exclusive.add_argument(
+        '--npv', type=int,
+        help='Number of seeds per voxel.')
+    seed_sub_exclusive.add_argument(
+        '--nt', type=int,
+        help='Total number of seeds to use.')
+
+    p.add_argument(
+        '--sphere', choices=sorted(SPHERE_FILES.keys()),
+        default='symmetric724',
```

Comment:
Default should probably change. It is `repulsion` in the PFT code, and Dipy is also changing.